File: backend-py/req/get_ausnahmen.py
import requests
from dataclasses import dataclass
from typing import List
from config import STRAPI_URL

def get_ausnahmen(jwt: str) -> List[dict]:
    url = f"{STRAPI_URL}/ausnahmen"
    headers = {
        "Authorization": f"Bearer {jwt}"
    }
    response = requests.get(url, headers=headers)
    response.raise_for_status()  # Raise an exception for HTTP errors
    res = response.json()
    data = res['data']
    attributeList = []
    for ausnahme in data:
        attributes = ausnahme['attributes']
        attributeList.append(attributes)
    return attributeList

File: src-tauri/src/req/get_ausnahmen.rs
use serde::Deserialize;
use crate::config::Config;

#[derive(Deserialize, Debug)]
pub struct AusnahmenData {
    pub data: Vec<IdAtrAusnahmen>,
}

#[derive(Deserialize, Debug)]
pub struct IdAtrAusnahmen {
    pub id: i16,
    pub attributes: Ausnahmen,
}

#[derive(Deserialize, Debug)]
#[allow(non_snake_case)]
pub struct Ausnahmen {
    pub Barcode: String,
    pub Bedeutung: String,
}

// get all exceptions from the database
#[tokio::main]
pub async fn get_ausnahmen(jwt: &str) -> Result<Vec<Ausnahmen>, reqwest::Error> {
    let config = Config::new();
    let url = format!("{}/api/ausnahmen", config.api.strapi);
    let client = reqwest::Client::new();

    let res = client
        .get(&url)
        .header("Authorization", format!("Bearer {}", jwt))
        .send()
        .await?
        .json::<AusnahmenData>()
        .await?;

    let mut ausnahmen = Vec::new();

    for ausnahme in res.data {
        ausnahmen.push(ausnahme.attributes);
    }

    Ok(ausnahmen)
}